frontend2/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { forwardRef, useMemo, useRef, useState } from "react";
+import FormLabel from "./elements/FormLabel";
+import FormError from "./elements/FormError";
+import type { Maybe } from "../utils/utilTypes";
+
+interface FileUploadProps extends React.ComponentPropsWithoutRef<"input"> {
+  label?: string;
+  required?: boolean;
+  className?: string;
+  errorMessage?: string;
+  fileInputHelp?: string;
+}
+
+const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
+  function FileUpload(
+    {
+      label,
+      required = false,
+      className = "",
+      errorMessage,
+      fileInputHelp,
+      ...rest
+    },
+    ref,
+  ) {
+    const invalid = errorMessage !== undefined;
+    const disabledClassName = `${invalid ? "ring-red-500" : ""}
+    ${rest.disabled === true ? "bg-gray-400/20 text-gray-600" : ""}`;
+
+    const fileUploadRef = useRef<HTMLInputElement>(null);
+
+    const [fileName, setFileName] = useState<Maybe<string>>();
+
+    useMemo(() => {
+      setFileName((fileUploadRef.current?.files ?? [])[0]?.name);
+    }, [fileUploadRef.current?.files]);
+
+    return (
+      <div className={`relative ${invalid ? "mb-1" : ""} ${className}`}>
+        {label !== undefined && <FormLabel label={label} />}
+        <label className="flex max-w-md flex-row items-center">
+          <input
+            aria-invalid={errorMessage !== undefined ? "true" : "false"}
+            type="file"
+            className="hidden"
+            ref={ref}
```

Comment:
I need this to be `fileUploadRef` so I can get the file name to display (the `forwardedRef` doesn't have this) :(